qa/rpc-tests/test_framework/mininode.py

```diff
@@ -1126,11 +1126,14 @@ def calc_sha256(self):
         if self.nVersion >= 5:
             from . import zip244
             txid = zip244.txid_digest(self)
+            self.auth_digest = zip244.auth_digest(self)
         else:
             txid = hash256(self.serialize())
+            self.auth_digest = b'\xFF'*32
```

Comment:
I suggest moving this to `zip244`:
```suggestion
        from . import zip244
        (txid, self.auth_digest) = zip244.wtxid(self)
```

where `zip244` has:
```python
"""
Return a pair of `(txid, auth_digest)`, where for v4 transactions, `auth_digest`
is the placeholder value specified in <https://zips.z.cash/zip-0244#authorizing-data-commitment>.
"""
def wtxid(tx):
    if self.nVersion >= 5:
        return (txid_digest(tx), auth_digest(tx))
    else:
        return (hash256(tx.serialize()), b'\xFF'*32)
```

Reply:
We can do this in a subsequent cleanup PR post-hotfix.

Reply:
Here's a patch that I tested and works that we can merge in a later PR:
<details>
<summary>patch diff</summary>

```diff
index 2c51f4421..5b16b0acf 100755
--- a/qa/rpc-tests/test_framework/mininode.py
+++ b/qa/rpc-tests/test_framework/mininode.py
@@ -1123,13 +1123,8 @@ class CTransaction(object):
         self.calc_sha256()
 
     def calc_sha256(self):
-        if self.nVersion >= 5:
-            from . import zip244
-            txid = zip244.txid_digest(self)
-            self.auth_digest = zip244.auth_digest(self)
-        else:
-            txid = hash256(self.serialize())
-            self.auth_digest = b'\xFF'*32
+        from .zip244 import wtxid
+        txid, self.auth_digest = wtxid(self)
         if self.sha256 is None:
             self.sha256 = uint256_from_str(txid)
         self.hash = encode(txid[::-1], 'hex_codec').decode('ascii')
diff --git a/qa/rpc-tests/test_framework/zip244.py b/qa/rpc-tests/test_framework/zip244.py
index f9cd5ceee..d4b03ade8 100644
--- a/qa/rpc-tests/test_framework/zip244.py
+++ b/qa/rpc-tests/test_framework/zip244.py
@@ -15,7 +15,12 @@ import struct
 
 from pyblake2 import blake2b
 
-from .mininode import ser_string, ser_uint256
+from .mininode import (
+    hash256,
+    ser_string,
+    ser_uint256,
+)
+
 from .script import (
     SIGHASH_ANYONECANPAY,
     SIGHASH_NONE,
@@ -292,3 +297,13 @@ def txin_sig_digest(tx, txin):
     digest.update(struct.pack('<Q', txin.amount))
     digest.update(struct.pack('<I', tx.vin[txin.nIn].nSequence))
     return digest.digest()
+
+"""
+Return a pair of `(txid, auth_digest)`, where for v4 transactions, `auth_digest` is the
+placeholder value specified in <https://zips.z.cash/zip-0244#authorizing-data-commitment>.
+"""
+def wtxid(tx):
+    if tx.nVersion >= 5:
+        return (txid_digest(tx), auth_digest(tx))
+    else:
+        return (hash256(tx.serialize()), b'\xFF'*32)
```
</details>